aptos-move/aptos-gas-algebra/src/algebra.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+pub use move_core_types::gas_algebra::*;
+
+/***************************************************************************************************
+ * Units & Quantities
+ *
+ **************************************************************************************************/
+/// Unit of abstract value size -- a conceptual measurement of the memory space a Move value occupies.
+pub enum AbstractValueUnit {}
+
+pub type AbstractValueSize = GasQuantity<AbstractValueUnit>;
+
+pub type InternalGasPerAbstractValueUnit = GasQuantity<UnitDiv<InternalGasUnit, AbstractValueUnit>>;
+
+pub type AbstractValueSizePerArg = GasQuantity<UnitDiv<AbstractValueUnit, Arg>>;
+
+/// Unit of (external) gas.
+pub enum GasUnit {}
+
+/// Unit of the Aptos network's native coin.
+pub enum APT {}
+
+/// Alternative unit of the Aptos network's native coin. 1 Octa = 10^-8 Aptos coins.
+pub enum Octa {}
+
+pub type Gas = GasQuantity<GasUnit>;
+
+pub type GasScalingFactor = GasQuantity<UnitDiv<InternalGasUnit, GasUnit>>;
```

Comment:
Because there is an upstream gas algebra module in `third_party/move/move-core/types` defining the basics. `InternalGas` is also being referenced in the Move VM, even though it's merely used as a medium for handling native function costs.